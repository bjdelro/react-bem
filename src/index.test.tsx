import { render } from 'enzyme'
import * as React from 'react'

import createBEMBlock from './createBEMBlock'
import createBEMElement from './createBEMElement'

describe('react-bem HOCs', () => {

	class Foo extends React.Component {
		render() {
			return (
				<div children={this.props.children} />
			)
		}
	}

	const FooBlock = createBEMBlock(Foo)

	class Item extends React.Component {
		render() {
			return (
				<div />
			)
		}
	}

	const ItemElement = createBEMElement(Item)

	it('renders an element within a block within a block', () => {
		expect(render(
			<FooBlock block="b1">
				<ItemElement element="e1" />
				<FooBlock block="b2">
					<ItemElement element="e2" />
				</FooBlock>
			</FooBlock>
		)).toMatchSnapshot()
	})
})
