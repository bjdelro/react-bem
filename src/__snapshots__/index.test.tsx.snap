// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-bem HOCs renders an element within a block within a block 1`] = `
<div
  class="b1"
>
  <div
    class="b1__e1"
  />
  <div
    class="b2"
  >
    <div
      class="b2__e2"
    />
  </div>
</div>
`;
